### PR TITLE
[GOBBLIN-731] Make deserialization of FlowSpec more robust

### PR DESCRIPTION
**Pull request by autumnust**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [ ] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/731) issues and references them in the PR title.

### Description

Catching exception when loading `FlowSpec` from FileSystem and make sure that even some of spec files are problematic, the rest of them will still be successfully deserialized.

Adding unit test for that. 

Removing some unnecessary modifiers while reading through the code. 

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by autumnust** [comment]
@sv2000 @yukuai518  Can you take a look ? Thanks.

File: gobblin-runtime/src/test/java/org/apache/gobblin/spec_catalog/FlowCatalogTest.java [hunk 1]
```diff
@@ -17,15 +17,25 @@
 
 package org.apache.gobblin.spec_catalog;
 
+import de.flapdoodle.embed.process.io.file.Files;
```
Comment (arjun4084346) [inline gobblin-runtime/src/test/java/org/apache/gobblin/spec_catalog/FlowCatalogTest.java]: You did not mean this import

File: gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_store/FSSpecStore.java [hunk 2]
```diff
@@ -238,13 +238,25 @@ public Spec getSpec(URI specUri, String version) throws IOException, SpecNotFoun
...
   private void getSpecs(Path directory, Collection<Spec> specs) throws IOException {
     FileStatus[] fileStatuses = fs.listStatus(directory);
     for (FileStatus fileStatus : fileStatuses) {
       if (fileStatus.isDirectory()) {
         getSpecs(fileStatus.getPath(), specs);
       } else {
-        specs.add(readSpecFromFile(fileStatus.getPath()));
+        try {
+          specs.add(readSpecFromFile(fileStatus.getPath()));
+        } catch (IOException ioe) {
```
Comment (sv2000) [inline gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_store/FSSpecStore.java]: Catching IOException may not be enough. SerializationUtils.deserialize() throws a SerializationException which extends RuntimeException.

File: gobblin-runtime/src/test/java/org/apache/gobblin/spec_catalog/FlowCatalogTest.java [hunk 3]
```diff
@@ -117,6 +127,45 @@ public void cleanUp() throws Exception {
     }
   }
 
+  /**
+   * Make sure that when there's on spec failed to be deserialized, the rest of spec in specStore can
+   * still be taken care of.
+   */
+  @Test
+  public void testGetSpecRobustness() throws Exception {
```
Comment (sv2000) [inline gobblin-runtime/src/test/java/org/apache/gobblin/spec_catalog/FlowCatalogTest.java]: Can the test include reading a 0 byte spec from the specStore? Also, can the readSpecFromFile() method in TestSpecStore call SerializationUtils.deserialize() on the 0 byte file? This way we can have it throw a SerializationException.

**Pull request closed without merging**